common/persistence/sql/sqlplugin/sqlite/plugin.go

```diff
@@ -112,9 +123,38 @@ func (p *plugin) createDBConnection(
 	// Maps struct names in CamelCase to snake without need for db struct tags.
 	db.MapperFunc(strcase.ToSnake)
 
+	// only suitable for in memory databases!
+	if _, ok := cfg.ConnectAttributes["setup"]; ok {
```

Comment:
Heads up we pass connection attributes more or less transparently into the sqlite driver here: https://github.com/temporalio/temporal/blob/c38d169e85eb1729651fdeb2c2bb0265b002ba57/common/persistence/sql/sqlplugin/sqlite/plugin.go#L176-L187

The current driver may not be picky about it but I think we should avoid passing in non-standard sqlite config options. How about checking if the driver is using memory mode instead? When persisting to a file we can use the same migration patterns we do for postgres/mysql/cassandra backends.

```suggestion
	if mode := cfg.ConnectAttributes["mode"]; mode == "memory" {
```

Reply:
I have not seen a notification about this question. Ugh. I want to leave the dedicated flag since it makes possible to run bootstrap on sqlite file database (which is handy). But now this flag will be removed.

Reply:
@jlegrone Also, I tried https://github.com/mattn/go-sqlite3 and I do not see a quick path to integrate it here without adding a few DB layer abstractions for sqlx compatibility. It might be easier and more effective to build a new DB type for in-memory use.